backend/company-agent/src/main/java/org/hyperledger/oa/impl/aries/SchemaService.java

```diff
@@ -57,22 +58,28 @@
     // CRUD Methods
 
     public SchemaAPI addSchema(@NonNull String schemaId, @Nullable String label) {
+    	return addSchema(schemaId, label, false); 
+    }
+    
+    public SchemaAPI addSchema(@NonNull String schemaId, @Nullable String label, boolean isReadOnly) {
         SchemaAPI result = null;
         String sId = StringUtils.strip(schemaId);
         final CredentialType credType = CredentialType.fromSchemaId(sId);
-        if (schemaRepo.findByType(credType).isPresent()) {
-            throw new WrongApiUsageException("Schema with type: " + credType + " already exists.");
+
+        // schemes of type other than OTHER should only exist once
+        if (!credType.equals(CredentialType.OTHER) && schemaRepo.findByType(credType).isPresent()) {
```

Comment:
Still not sure about that one. What makes the most sense is to make the schemaId unique, to prevent adding the same schema twice. This is now possible and then the findBySchemaId() method will throw an exception.